src/julia.jl

```diff
@@ -33,4 +33,139 @@ function multireshape(parent::AbstractArray, dimss::Array{Tuple{Vararg{Int}}})
     multireshape(parent, dimss...)
 end
 
+
+"""
+    pairwise(f::Function, a; kwargs...)
+
+Apply the function `f` to all pairwise combinations of elements from `a`. Set
+keyword `symmetric = true` to only calculate `f(x,y)` when `f(x,y) == f(y,x)`.
+"""
+function pairwise(f::Function, a; symmetric = false)
+    n = length(a)
+    f = tryfunc(f, a[1], a[1])
+    first = f(a[1], a[1])
+    r = Matrix{typeof(first)}(n, n)
+    pairwise!(f, a, r; symmetric = symmetric)
+    r
+end
+
+"""
+    pairwise(f::Function, a, output; kwargs...)
+
+Apply the function `f` to all pairwise combinations of elements from `a`, and
+return a `Vector` of the results if `output = :Vector`. The Vector form is
```

Comment:
If you are passing an argument where you don't actually look at the value of the argument (you only look at the type), then you should pass an instance of a singleton type (a type that has only one instance).  For example, `Type{Vector}` is a singleton type with only one instance, `Vector`.